src/wms/src/test/java/org/geoserver/wms/map/RenderedImageMapOutputFormatTest.java

```diff
@@ -350,6 +350,154 @@ public void testInterpolations() throws IOException, IllegalFilterException, Exc
                 getPixelColor(imageBicubic, 300, 300).getRGB());
     }
 
+    @Test
+    public void testInterpolationFromLayerConfig() throws IOException, IllegalFilterException, Exception {
+        final Catalog catalog = getCatalog();
+
+        LayerInfo layerInfo = catalog.getLayerByName(MockData.TASMANIA_DEM.getLocalPart());
+
+        MapLayerInfo mapLayer = new MapLayerInfo(layerInfo);
+        assertNull(layerInfo.getDefaultInterpolationMethod());
+
+        Envelope env = layerInfo.getResource().boundingBox();
+        double shift = env.getWidth() / 6;
+
+        env = new Envelope(env.getMinX() - shift, env.getMaxX() + shift, env.getMinY() - shift,
+                env.getMaxY() + shift);
+
+        // set Nearest Neighbor interpolation on layer
+        GetMapRequest request = new GetMapRequest();
+        request.setFormat(getMapFormat());
+        request.setLayers(Arrays.asList(mapLayer));
+
+        layerInfo.setDefaultInterpolationMethod("Nearest");
+        assertEquals("Nearest",
+                request.getLayers().get(0).getLayerInfo().getDefaultInterpolationMethod());
+        assertTrue(request.getInterpolations().isEmpty());
+
+        WMSMapContent map = createWMSMap(env);
+        map.setRequest(request);
+
+        RenderedImageMap imageMap = this.rasterMapProducer.produceMap(map);
```

Comment:
One non binding observation (no need for change, it's a nice to have). You could extend WebMap a bit to allow access to the WebMapContent, and then use it to check if the right interpolation has been set in the layer, seems a bit better than just chechking that "something changed" (which is fair by the way, one cannot expect  particular value in the output pixels as JAI could be pure java, native, jai-ext, or OSX one, and they would return slight different values)